samples/MQTTSN/src/transport.c

```diff
@@ -0,0 +1,167 @@
+/*******************************************************************************
+ * Copyright (c) 2014 IBM Corp.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *   http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * Contributors:
+ *    Ian Craggs - initial API and implementation and/or initial documentation
+ *    Sergio R. Caprile - "commonalization" from prior samples and/or documentation extension
+ *******************************************************************************/
+
+#include <sys/types.h>
+
+#if !defined(SOCKET_ERROR)
+/** error in socket operation */
+#define SOCKET_ERROR -1
+#endif
+
+#if defined(WIN32)
+/* default on Windows is 64 - increase to make Linux and Windows the same */
+#define FD_SETSIZE 1024
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#define MAXHOSTNAMELEN 256
+#define EAGAIN WSAEWOULDBLOCK
+#define EINTR WSAEINTR
+#define EINVAL WSAEINVAL
+#define EINPROGRESS WSAEINPROGRESS
+#define EWOULDBLOCK WSAEWOULDBLOCK
+#define ENOTCONN WSAENOTCONN
+#define ECONNRESET WSAECONNRESET
+#define ioctl ioctlsocket
+#define socklen_t int
+#else
+#define INVALID_SOCKET SOCKET_ERROR
+#include <arpa/inet.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <netinet/tcp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/param.h>
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <unistd.h>
+#endif
+
+#if defined(WIN32)
+#include <Iphlpapi.h>
+#else
+#include <net/if.h>
+#include <sys/ioctl.h>
+#endif
+
+#ifndef SRC_PORT
+#define SRC_PORT 1234
```

Comment:
I take my work back ... lets not define default value here.... Please pass this through cmake.

You can delete these lines here and set the SRC_PORT in cmake.